packages/hooks/src/use-field-id/use-field-id.ts

```diff
@@ -3,8 +3,9 @@ import { getFieldId } from '@commercetools-uikit/utils';
 
 type CreateIdFn = () => string;
 
-const useFieldId = (id: string, createIdFn: CreateIdFn): string => {
-  const [internalId, setId] = React.useState<string>(id);
+// @ts-ignore
+const useFieldId = (id?: string, createIdFn: CreateIdFn): string => {
```

Comment:
this has to be optional without updating the signature.
`@ts-ignore` is added, since we have a requirement that a required param shouldn't follow an optional. changing this is risk to breaking change.

will update the issue for v13.

Reply:
Can you remind me why this should be optional?

Reply:
Some components made `props.id` optional, but still pass it to `useFieldId`

Reply:
sounds good, let's try that.

Reply:
https://github.com/commercetools/ui-kit/pull/1901/commits/4f03a25af6b9ceda96fb5728dd68ae770d756cc9